Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -1065,107 +1065,118 @@ theorem comp_familyOfBFamily {o} (f : ∀ a < o, α) (g : α → β) :
+    (∀ i, f i ≤ a) → iSup f ≤ a :=
+  ciSup_le'
+
+-- TODO: generalize to conditionally complete lattices.
+protected theorem lt_ciSup {ι : Type u} {f : ι → Ordinal.{max u v}} {a} :
+    a < iSup f ↔ ∃ i, a < f i := by
+  rw [← not_iff_not]
+  simpa using Ordinal.ciSup_le_iff
+
+-- TODO: generalize to Sup, take `bddAbove` as an assumption.
+
+theorem ne_ciSup_iff_lt_ciSup {ι : Type u} {f : ι → Ordinal.{max u v}} :
+    (∀ i, f i ≠ iSup f) ↔ ∀ i, f i < iSup f :=
+  ⟨fun hf _ => lt_of_le_of_ne (Ordinal.le_ciSup _ _) (hf _), fun hf _ => ne_of_lt (hf _)⟩
+
+theorem ciSup_not_succ_of_ne_ciSup {ι : Type u} {f : ι → Ordinal.{max u v}}
+    (hf : ∀ i, f i ≠ iSup f) {a} (hao : a < iSup f) : succ a < iSup f := by
   by_contra! hoa
-  exact
-    hao.not_le (sup_le fun i => le_of_lt_succ <| (lt_of_le_of_ne (le_sup _ _) (hf i)).trans_le hoa)
+  exact hao.not_le (Ordinal.ciSup_le fun i => le_of_lt_succ <|
+    (lt_of_le_of_ne (Ordinal.le_ciSup _ _) (hf i)).trans_le hoa)
 
-@[simp]
-theorem sup_eq_zero_iff {ι : Type u} {f : ι → Ordinal.{max u v}} :
-    sup.{_, v} f = 0 ↔ ∀ i, f i = 0 := by
+-- TODO: generalize to conditionally complete lattices.
+theorem ciSup_eq_zero_iff {ι : Type u} {f : ι → Ordinal.{max u v}} :
+    iSup f = 0 ↔ ∀ i, f i = 0 := by
   refine
     ⟨fun h i => ?_, fun h =>
-      le_antisymm (sup_le fun i => Ordinal.le_zero.2 (h i)) (Ordinal.zero_le _)⟩
+      le_antisymm (Ordinal.ciSup_le fun i => Ordinal.le_zero.2 (h i)) (Ordinal.zero_le _)⟩
   rw [← Ordinal.le_zero, ← h]
-  exact le_sup f i
-
-theorem IsNormal.sup {f : Ordinal.{max u v} → Ordinal.{max u w}} (H : IsNormal f) {ι : Type u}
-    (g : ι → Ordinal.{max u v}) [Nonempty ι] : f (sup.{_, v} g) = sup.{_, w} (f ∘ g) :=
-  eq_of_forall_ge_iff fun a => by
-    rw [sup_le_iff]; simp only [comp]; rw [H.le_set' Set.univ Set.univ_nonempty g] <;>
-      simp [sup_le_iff]
-
-@[simp]
-theorem sup_empty {ι} [IsEmpty ι] (f : ι → Ordinal) : sup f = 0 :=
+  exact Ordinal.le_ciSup f i
+
+-- TODO: generalize universes
+theorem IsNormal.iSup {f : Ordinal.{max u v} → Ordinal.{max u w}} (H : IsNormal f) {ι : Type u}
+    (g : ι → Ordinal.{max u v}) [Nonempty ι] : f (⨆ i, g i) = ⨆ i, f (g i) := by
+  apply eq_of_forall_ge_iff
+  intro a
+  rw [H.le_set' Set.univ Set.univ_nonempty g]
+  · rw [Ordinal.ciSup_le_iff]
+    simp
+  · intro o
+    rw [Ordinal.ciSup_le_iff]
+    simp
+
+@[deprecated ciSup_of_empty (since := "2024-08-14")]
+theorem iSup_empty {ι} [IsEmpty ι] (f : ι → Ordinal) : iSup f = 0 :=
   ciSup_of_empty f
 
-@[simp]
-theorem sup_const {ι} [_hι : Nonempty ι] (o : Ordinal) : (sup fun _ : ι => o) = o :=
+@[deprecated ciSup_const (since := "2024-08-14")]
+theorem iSup_const {ι} [_hι : Nonempty ι] (o : Ordinal) : (⨆ _ : ι, o) = o :=
   ciSup_const
 
-@[simp]
-theorem sup_unique {ι} [Unique ι] (f : ι → Ordinal) : sup f = f default :=
+@[deprecated ciSup_unique (since := "2024-08-14")]
+theorem iSup_unique {ι} [Unique ι] (f : ι → Ordinal) : iSup f = f default :=
```

Comment:
There's no point deprecating a theorem you're renaming.
```suggestion
@[deprecated ciSup_unique (since := "2024-08-14")]
theorem sup_unique {ι} [Unique ι] (f : ι → Ordinal) : iSup f = f default :=
```
Also please add deprecated aliases for everything you're removing in this file

Reply:
I think that should be all of them.